internal/backends/qt/qt_window.rs

```diff
@@ -734,7 +735,82 @@ impl ItemRenderer for QtItemRenderer<'_> {
                     QString to_elide = QStringView(string).mid(last_line_begin, last_line_size).trimmed() % QStringView(QT_UNICODE_LITERAL("…"));
                     elided += fm.elidedText(to_elide, Qt::ElideRight, rect.width());
                 }
+            }
+
+            if (!stroke_visible) {
+                int flags = alignment;
+                if (wrap)
+                    flags |= Qt::TextWordWrap;
+
+                (*painter)->setFont(font);
+                (*painter)->setBrush(Qt::NoBrush);
+                (*painter)->setPen(QPen(fill_brush, 0));
                 (*painter)->drawText(rect, flags, elided);
+            } else {
+                QTextDocument document(elided);
```

Comment:
I'd prefer if this code used QTextLayout for drawing - that's more lightweight than using an entire QTextDocument (with a piece table, etc. behind it). But that could be a follow-up optimisation.

Reply:
This was the approach I initially considered as well, but as far as I could tell it would require me to manually implement wrapping. It’s certainly doable though, and I think it would resolve the performance issues I’m having with Qt specifically.